Tests/CryptoTests/ECDH/BoringSSL/secpECDH_Runner_boring.swift

```diff
@@ -25,34 +25,35 @@ import XCTest
-            return try! Array(hexString: hexStringFromVector)
+            return try orFail(file: file, line: line) { try Array(hexString: hexStringFromVector) }
         }
 
         for testVector in group.tests {
             do {
-                let pkBytes = try Array(hexString: testVector.publicKey)
-                let publicKey = try PrivKey.PublicKey(derBytes: pkBytes, curve: Curve.self)
+                let pkBytes = try orFail(file: file, line: line) { try Array(hexString: testVector.publicKey) }
+                let publicKey = try orFail(file: file, line: line) { try PrivKey.PublicKey(derBytes: pkBytes, curve: Curve.self) }
 
                 var privateBytes = [UInt8]()
-                privateBytes = padKeyIfNecessary(vector: testVector.privateKey, curveDetails: curve)
+                privateBytes = try orFail(file: file, line: line) { try padKeyIfNecessary(vector: testVector.privateKey, curveDetails: curve) }
 
-                let privateKey = try PrivKey(rawRepresentation: privateBytes)
+                let privateKey = try orFail(file: file, line: line) { try PrivKey(rawRepresentation: privateBytes) }
 
-                let result = try privateKey.sharedSecretFromKeyAgreement(with: publicKey as! PrivKey.P)
+                let agreement = try unwrap(publicKey as? PrivKey.P, file: file, line: line)
+                let result = try orFail(file: file, line: line) { try privateKey.sharedSecretFromKeyAgreement(with: agreement) }
 
-                let expectedResult = try Array(hexString: testVector.shared)
+                let expectedResult = try orFail(file: file, line: line) { try Array(hexString: testVector.shared) }
 
-                XCTAssertEqual(Array(result.ss), Array(expectedResult))
+                XCTAssertEqual(Array(result.ss), Array(expectedResult), file: file, line: line)
             } catch ECDHTestErrors.PublicKeyFailure {
-                XCTAssert(testVector.flags.contains("CompressedPoint") || testVector.result == "invalid" || testVector.flags.contains("InvalidPublic") || testVector.flags.contains("InvalidAsn"))
+                XCTAssert(testVector.flags.contains("CompressedPoint") || testVector.result == "invalid" || testVector.flags.contains("InvalidPublic") || testVector.flags.contains("InvalidAsn"), file: file, line: line)
             } catch ECDHTestErrors.ParseSPKIFailure {
-                XCTAssert(testVector.flags.contains("InvalidAsn") || testVector.flags.contains("UnnamedCurve"))
+                XCTAssert(testVector.flags.contains("InvalidAsn") || testVector.flags.contains("UnnamedCurve"), file: file, line: line)
             } catch {
                 if testVector.result == "valid" {
-                    XCTAssert(testVector.tcId == 31 || testVector.tcId == 20 || testVector.tcId == 25)
+                    XCTAssert(testVector.tcId == 31 || testVector.tcId == 20 || testVector.tcId == 25, file: file, line: line)
                 }
             }
```

Comment:
This test is not quite the same. Some errors were accepted here, and now they all trigger test failure.

Reply:
Oops, I had overzealously wrapped some things that were actually supposed to throw during testing. Fixed now.